# set the install directory directly
# looking it up does not work. probably because of the symlinks.
Index: pycharm-2017.2/bin/pycharm.sh
===================================================================
--- pycharm-2017.2.orig/bin/pycharm.sh	2017-07-27 17:48:37.676712032 +0200
+++ pycharm-2017.2/bin/pycharm.sh	2017-07-27 17:51:57.518696833 +0200
@@ -50,10 +50,11 @@
   done
 fi
 
-cd "`dirname "$SCRIPT_LOCATION"`"
-IDE_BIN_HOME=`pwd`
-IDE_HOME=`dirname "$IDE_BIN_HOME"`
-cd "$OLDPWD"
+IDE_HOME=/usr/share/pycharm
+IDE_BIN_HOME=/usr/share/pycharm/bin
+
+echo "IDE_HOME: " $IDE_HOME
+echo "IDE_BIN_HOME: " $IDE_BIN_HOME
 
 # ---------------------------------------------------------------------
 # Locate a JDK installation directory which will be used to run the IDE.
@@ -91,52 +92,14 @@
   JDK="$JAVA_HOME"
 fi
 
-if [ -z "$JDK" ]; then
-  JDK_PATH=`which java`
-
-  if [ -n "$JDK_PATH" ]; then
-    if [ "$OS_TYPE" = "FreeBSD" -o "$OS_TYPE" = "MidnightBSD" ]; then
-      JAVA_LOCATION=`JAVAVM_DRYRUN=yes java | "$GREP" '^JAVA_HOME' | "$CUT" -c11-`
-      if [ -x "$JAVA_LOCATION/bin/java" ]; then
-        JDK="$JAVA_LOCATION"
-      fi
-    elif [ "$OS_TYPE" = "SunOS" ]; then
-      JAVA_LOCATION="/usr/jdk/latest"
-      if [ -x "$JAVA_LOCATION/bin/java" ]; then
-        JDK="$JAVA_LOCATION"
-      fi
-    elif [ "$OS_TYPE" = "Darwin" ]; then
-      JAVA_LOCATION=`/usr/libexec/java_home`
-      if [ -x "$JAVA_LOCATION/bin/java" ]; then
-        JDK="$JAVA_LOCATION"
-      fi
-    fi
-  fi
-
-  if [ -z "$JDK" -a -x "$READLINK" -a -x "$XARGS" -a -x "$DIRNAME" ]; then
-    JAVA_LOCATION=`"$READLINK" -f "$JDK_PATH"`
-    case "$JAVA_LOCATION" in
-      */jre/bin/java)
-        JAVA_LOCATION=`echo "$JAVA_LOCATION" | "$XARGS" "$DIRNAME" | "$XARGS" "$DIRNAME" | "$XARGS" "$DIRNAME"`
-        if [ ! -d "$JAVA_LOCATION/bin" ]; then
-          JAVA_LOCATION="$JAVA_LOCATION/jre"
-        fi
-        ;;
-      *)
-        JAVA_LOCATION=`echo "$JAVA_LOCATION" | "$XARGS" "$DIRNAME" | "$XARGS" "$DIRNAME"`
-        ;;
-    esac
-    if [ -x "$JAVA_LOCATION/bin/java" ]; then
-      JDK="$JAVA_LOCATION"
-    fi
-  fi
-fi
+JDK=`update-alternatives --list java | grep java-8 | sed 's,jre/bin/java,,'`
 
 JAVA_BIN="$JDK/bin/java"
 if [ -z "$JDK" -o ! -x "$JAVA_BIN" ]; then
   message "No JDK found. Please validate either PYCHARM_JDK, JDK_HOME or JAVA_HOME environment variable points to valid JDK installation."
   exit 1
 fi
+echo "JAVA_BIN: " $JAVA_BIN
 
 VERSION_LOG=`"$MKTEMP" -t java.version.log.XXXXXX`
 JAVA_TOOL_OPTIONS= "$JAVA_BIN" -version 2> "$VERSION_LOG"
